src/3dsystem/matrix.c

```diff
@@ -45,12 +47,19 @@ void phd_GenerateW2V(PHD_3DPOS *viewpos)
 
 void phd_PushMatrix(void)
 {
+    if (g_PhdMatrixPtr + 1 - m_MatrixStack >= MAX_MATRICES) {
+        Draw_PrintRoomNumStack();
+        Shell_ExitSystem("Push matrix stack overflow.");
+    }
     g_PhdMatrixPtr++;
     g_PhdMatrixPtr[0] = g_PhdMatrixPtr[-1];
 }
 
 void phd_PushUnitMatrix(void)
 {
+    if (g_PhdMatrixPtr + 1 - m_MatrixStack >= MAX_MATRICES) {
+        Shell_ExitSystem("Push unit matrix stack overflow.");
```

Comment:
Missing call to `Draw_PrintRoomNumStack`. I don't think the log needs to differentiate between unit matrices and duplicate matrices, IMO not.

Reply:
I still see the same log message